### ISSUE TITLE
Indentation in dbms output is odd, when multiple tests are run 

### ISSUE THREAD

**Issue by durrani** [utPLSQL/utPLSQL#482]
declare
begin
ut.run('ut_my_pkg.test01');
ut.run('ut_my_pkg.test02');
ut.run('ut_my_pkg.test03');
end;
/
`
Expected output:
<pre>
ut_my_pkg
  test01
  *output from test01*
Finished in .78 seconds
1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  

ut_my_pkg
  test02
  *output from test02*
Finished in .78 seconds
1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  

ut_my_pkg
  test03
  *output from test03*
Finished in .78 seconds
1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  
</pre>

Actual output:
<pre>
ut_my_pkg
  test03
  ut_my_pkg
    test02
    ut_my_pkg
      test01
      *output from test01*
      Finished in .78 seconds
        1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  
        *output from test02*
   Finished in .78 seconds
  1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  
  *output fom test 03'
Finished in .78 seconds
1 tests, 0 failed, 0 errored, 0 disabled 0 warning(s)  
</pre>
`

**Comment by jgebal** [comment]
Confirmed the behavior on version 3.0.3.
It is odd indeed.

**Comment by jgebal** [comment]
The use-case is odd too by the wy.
I would suggest using:
```sql
begin
  ut.run(ut_varchar2_list('ut_my_pkg.test01','ut_my_pkg.test02','ut_my_pkg.test03'));
end;
/
```

**Comment by jgebal** [comment]
Afer some thinking I know why it's happening.
The output from DBMS_OUPTUT is not consumed by clent (SQL Developer/TOAD/*SQLPlus) untill the execution of the PL/SQL block completes.

Second call to `ut.run` intercepts all outputs from dbms_output that were not processed before the call.
The same happens to third call.

This is done by procedure `save_dbms_output` in ut_executable.
The reason behind this is that we want to capture and display all things that were placed in dbms_output buffer by unit test or by tested code.

The reporter does not indicate clearly that the inside data comes from intercepting `dbms_output`.

**possible solution**
Intercept all DBMS_OUTPUT before starting the run and keep it somewhere.
Add back the intercepted dbms_output after the run and before reporting the data back.

**Note**
You probably should invoke ut.run just once in one plsql block.

**Comment by durrani** [comment]
Thank you for your work.  Great tool.

I am using older version of utplsql.  3.0.3 is not in our development network as it has to go through security of our company :-(, hence can't use ut.run('test01', 'test02', 'test03');

Yes, as I said in the title of this issue, the output is from dbms_output.

**Comment by jgebal** [comment]
What version are you using then?
The syntax will work with 3.0.2 3.0.1 and 3.0.0 as far as I remember.

**Comment by jgebal** [comment]
Just pass in list of items as `ut_varchar2_list`

**Comment by durrani** [comment]
The version is 3.0.2 and  
ut.run(ut_varchar2_list('ut_my_pkg.test01','ut_my_pkg.test02','ut_my_pkg.test03'));  
does work for me as expected.

**Comment by jgebal** [comment]
I guess it would be nice to have utPLSQL intercept only dbms_output call made within the ut.run call.
Something to look at.
